src/Pirouette/Symbolic/Eval.hs

```diff
@@ -83,25 +85,26 @@ import qualified PureSMT
 
 data SymEvalSolvers lang = SymEvalSolvers
   { -- | Check whether a path is plausible
-    solvePathProblem :: CheckPathProblem lang -> Bool,
+    solvePathProblem :: CheckPathProblem lang -> IO Bool,
     -- | Check whether a certain property currently holds over a given path
-    solvePropProblem :: CheckPropertyProblem lang -> PruneResult
+    solvePropProblem :: CheckPropertyProblem lang -> IO PruneResult
   }
 
 data SymEvalEnv lang = SymEvalEnv
   { seeDefs :: PrtOrderedDefs lang,
     seeSolvers :: SymEvalSolvers lang,
+    seeAsyncSolver :: forall t res. Traversable t => t (SolverProblem lang res) -> IO (t res),
```

Comment:
For simplicity, I just added a binding in `SymEvalEnv` for solving problems asynchronously. The bindings inside `seeSolvers` now become a bit redundant.